### ISSUE TITLE
crash with strange format on xlsx file

### ISSUE THREAD

**Issue by gvollant** [damian-m-g/xlsx_drone#12]
I received and xlsx file with this strange format

[mini5.xlsx](https://github.com/damian-m-g/xlsx_drone/files/15354184/mini5.xlsx)

```
#,##0\ "gche";[Red]\-#,##0\ "gche"
```

xlsx_drone crash when xlsx_open is called (in get_formetted function, called by get_related_category)

I'll subit a PR with a string length verification

**Comment by gvollant** [comment]
@damian-m-g

**Comment by gvollant** [comment]
@damian-m-g what do you think?

**Comment by damian-m-g** [comment]
Hi Gilles, thanks for reporting the problem. I checked the XLSX, not in depth, but at least I found that _cell category (aka format code)_:

![image](https://github.com/damian-m-g/xlsx_drone/assets/1852056/b95b8dc2-ffb3-421c-9289-c53b851b08a9)

I couldn't yet debug the lib to reproduce the error, but I've tried to follow the problem by bare eye. I see no way, in which the length of the _cell category_ could cause a problem. So `get_related_category()` which in time calls `get_formatter()`, gets passed the length of the _cell category_:

![image](https://github.com/damian-m-g/xlsx_drone/assets/1852056/c017f863-ff02-4c87-a2f4-7b9ba32ad98f)

And then on `get_related_category()`, the code never pass an index out of the bounds of the _cell category_ length:

![image](https://github.com/damian-m-g/xlsx_drone/assets/1852056/37839ea5-567d-447b-949f-d71ffa59aa4d)

So your first check according to this analysis could be in vain. About your second check, I followed this specific case and it should work fine (without the fix). In conclusion, I need to set up my machine to debug this code, as it got formatted recently. This could even be a problem of the third hand libraries we are using (perhaps SXMLC), or something else. When I have some time available I'll do it.
